src/main/java/org/broadinstitute/hellbender/utils/variant/GATKVariantContextUtils.java

```diff
@@ -1017,28 +1004,46 @@ static private GenotypesContext fixGenotypesFromSubsettedAlleles(final Genotypes
     }
 
     /**
-     * Get the actual likelihoods indexes to use given the corresponding allele indexes
+     * For each genotype, get the likelihood indexes to use for the corresponding allele indexes
      *
      * @param originalVC           the original VariantContext
      * @param alleleIndexesToUse   the bitset representing the alleles to use (@see #getAlleleIndexBitset)
-     * @return a non-null List
+     * @return likelihoods indexes for each genotype
      */
-    private static List<Integer> getLikelihoodIndexes(final VariantContext originalVC, final boolean[] alleleIndexesToUse) {
+    private static List<List<Integer>> getLikelihoodIndexes(final VariantContext originalVC, final boolean[] alleleIndexesToUse) {
 
-        final List<Integer> result = new ArrayList<>(30);
+        final List<List<Integer>> likelihoodIndexesPerGenotype = new ArrayList<List<Integer>>(10);
 
-        // numLikelihoods takes total # of alleles. Use default # of chromosomes (ploidy) = 2
-        final int numLikelihoods = GenotypeLikelihoods.numLikelihoods(originalVC.getNAlleles(), DEFAULT_PLOIDY);
+        for (final Genotype g : originalVC.getGenotypes()) {
+            final int numLikelihoods = GenotypeLikelihoods.numLikelihoods(originalVC.getNAlleles(), g.getPloidy());
+            final List<Integer> likelihoodIndexes = new ArrayList<Integer>(30);
+            for ( int PLindex = 0; PLindex < numLikelihoods; PLindex++ ) {
+                // consider this entry only if all the alleles are good
+                if ( areAllAllelesUsed(PLindex, alleleIndexesToUse, g.getPloidy()) )
+                    likelihoodIndexes.add(PLindex);
+            }
+            likelihoodIndexesPerGenotype.add(likelihoodIndexes);
+        }
 
-        for ( int PLindex = 0; PLindex < numLikelihoods; PLindex++ ) {
-            final GenotypeLikelihoods.GenotypeLikelihoodsAllelePair alleles = GenotypeLikelihoods.getAllelePair(PLindex);
-            // consider this entry only if both of the alleles are good
-            if ( alleleIndexesToUse[alleles.alleleIndex1] && alleleIndexesToUse[alleles.alleleIndex2] ) {
-                result.add(PLindex);
+        return likelihoodIndexesPerGenotype;
+    }
+
+    /**
+     * For a PL index, check if all of the alleles are used
+     *
+     * @param PLindex               index into the number of likelihoods (@see #numLikelihoods)
+     * @param alleleIndexesToUse    the bitset representing the alleles to use (@see #getAlleleIndexBitset)
+     * @param ploidy                the number of chromosomes
+     * @return true if all of the alleles are used, false otherwise
+     */
+    private static boolean areAllAllelesUsed(final int PLindex, final boolean[] alleleIndexesToUse, final int ploidy) {
```

Comment:
replace this whole method by a simple `GenotypeLikelihoods.getAlleles(PLindex, ploidy).stream().allMatch(i -> alleleIndexesToUse[i]);`